shell/platform/linux/fl_renderer_x11.cc

```diff
@@ -53,16 +63,17 @@ static gboolean fl_renderer_x11_create_surfaces(FlRenderer* renderer,
                                                 EGLSurface* visible,
                                                 EGLSurface* resource,
                                                 GError** error) {
-  FlRendererX11* self = FL_RENDERER_X11(renderer);
-
-  if (!self->window) {
-    g_set_error(error, fl_renderer_error_quark(), FL_RENDERER_ERROR_FAILED,
-                "Can not create EGL surface: FlRendererX11::window not set");
+  GdkWindow* window = gtk_widget_get_window(
+      gtk_widget_get_toplevel(GTK_WIDGET(fl_renderer_get_view(renderer))));
```

Comment:
Why does this require the toplevel widget?

Reply:
It doesn't and that was actually causing a rendering bug. Fixed.